scripts/firebomb.particle

```diff
@@ -120,3 +120,113 @@ particles/weapons/firebomb/sub
 		period 0 - ~0%
 	}
 }
+
+// TODO: 0.54.1: rename as particles/weapons/firebomb/groundfire
+particles/weapons/flamer/floorfire
+{
+	// heat haze
+	ejector
+	{
+		particle
+		{
+			shader sync gfx/weapons/flamer/haze
+
+			displacement ~7 ~7 5 ~0
+
+			velocityType static
+			velocityDir linear
+			velocityMagnitude 0
+			velocity 0 0 1 ~60
+
+			accelerationType static
+			accelerationDir linear
+			accelerationMagnitude 10~10
+			acceleration 0 0 1 0
+
+			radius 0 5 45
+			alpha 0 .3~.3 0
+			rotation ~360 ~360 -
+			bounce .01
+
+			lifeTime 1500~50%
+		}
+
+		count infinite
+		delay 0
+		period 700 - ~50%
+	}
+
+	// smoke
+	ejector
+	{
+		particle
+		{
+			shader sync gfx/weapons/flamer/smoke
+
+			displacement ~5 ~5 5 ~0
+
+			velocityType static
+			velocityDir linear
+			velocityMagnitude 0
+			velocity 0 0 1 ~60
+
+			accelerationType static
+			accelerationDir linear
+			accelerationMagnitude 10~10
+			acceleration 0 0 1 0
+
+			radius 0 3 30~50%
+			alpha 0 .3~.3 0
+			rotation ~360 ~360 -
+			bounce .01
+
+			lifeTime 1500~50%
+		}
+
+		count infinite
+		delay 0
+		period 500 - ~50%
+	}
+
+	// flame
+	ejector
+	{
+		particle
+		{
+			shader sync gfx/weapons/firebomb/fire
+			displacement ~1 ~1 8 ~0
```

Comment:
In `displacement ~1 ~1 8 ~0`, `8` is the vertical displacement, positive is up, negative is down. `0` means the center of the image is on the ground.

@slipher adjusting this (relatively to the radius, which defines the image scaling) makes possible to prevent the particle from being seen from below.
Is this the part that supposedly fixes a bug about showing through walls?

Reply:
Also this fixes from seeing though ceiling, but particles may still be visible through wall because of horizontal displacement. Horizontal displacement is much less a problem with that “large flame” design, but that can still happen if by any luck the particle center is exactly spawned the closest possible to a wall, and that the wall is thin.